Kernel/Process.cpp

```diff
@@ -2735,53 +2738,63 @@ int Process::sys$reboot()
     return ESUCCESS;
 }
 
-int Process::sys$mount(const char* device_path, const char* mountpoint)
+int Process::sys$mount(const char* device_path, const char* mountpoint, const char* fstype)
 {
     if (!is_superuser())
         return -EPERM;
 
-    if (!validate_read_str(device_path) || !validate_read_str(mountpoint))
+    if (!validate_read_str(device_path) || !validate_read_str(mountpoint) || !validate_read_str(fstype))
         return -EFAULT;
 
-    dbg() << "mount: device " << device_path << " @ " << mountpoint;
+    dbg() << "mount " << fstype << ": device " << device_path << " @ " << mountpoint;
 
     auto custody_or_error = VFS::the().resolve_path(mountpoint, current_directory());
     if (custody_or_error.is_error())
         return custody_or_error.error();
 
     auto& mountpoint_custody = custody_or_error.value();
 
-    auto metadata_or_error = VFS::the().lookup_metadata(device_path, current_directory());
-    if (metadata_or_error.is_error())
-        return metadata_or_error.error();
+    RefPtr<FS> fs { nullptr };
```

Comment:
No need to explicitly initialize smart pointers;
```suggestion
    RefPtr<FS> fs;
```